extensions-core/histogram/src/main/java/org/apache/druid/query/aggregation/histogram/ApproximateHistogram.java

```diff
@@ -532,9 +534,9 @@ public ApproximateHistogram copy(ApproximateHistogram h)
   //approximate histogram solution using min heap to store location of min deltas
   protected ApproximateHistogram foldMin(
       ApproximateHistogram h,
-      float[] mergedPositions,
-      long[] mergedBins,
-      float[] deltas
+      @Nullable float[] mergedPositions,
```

Comment:
It is because this constructor:
https://github.com/apache/incubator-druid/blob/782863ed0f00fe199599b6bafc587afd929e0f14/extensions-core/histogram/src/main/java/org/apache/druid/query/aggregation/histogram/ApproximateHistogram.java#L475

Which makes this one `@Nullable`:
https://github.com/apache/incubator-druid/blob/782863ed0f00fe199599b6bafc587afd929e0f14/extensions-core/histogram/src/main/java/org/apache/druid/query/aggregation/histogram/ApproximateHistogram.java#L478-L485

Which trickles down to making `mergedPositions` nullable.

Reply:
This constructor is only used in tests and benchmarks.

Reply:
We can drop zero-sized arrays instead and drop in the `@Visible for testing`. Which is nicer in my opinion. Would love to get your opinion on this.

Reply:
Fixed this by removing the constructor entirely.

Reply:
Thanks